app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java

```diff
@@ -87,11 +91,33 @@ public void render(Chart chart, Map<String, Obs> latestObservations,
 
         mView.getSettings().setJavaScriptEnabled(true);
         mView.addJavascriptInterface(controllerInterface, "controller");
-        mView.setWebChromeClient(new WebChromeClient());
+
+        mView.setWebChromeClient(new WebChromeClient(){
+            @Override
+            public boolean onJsAlert(WebView view, String url, String message, final android.webkit.JsResult result)
+            {
+                new AlertDialog.Builder(mView.getContext())
+                        .setTitle(mView.getContext().getString(R.string.observation))
+                        .setMessage(message)
+                        .setPositiveButton(android.R.string.ok,
+                                new AlertDialog.OnClickListener()
+                                {
+                                    public void onClick(DialogInterface dialog, int which)
+                                    {
+                                        result.confirm();
+                                    }
+                                })
+                        .setCancelable(false)
+                        .create()
+                        .show();
+                return true;
+            }
+        });
+
         String html = new GridHtmlGenerator(chart, latestObservations, observations, orders,
                                             admissionDate, firstSymptomsDate).getHtml();
         mView.loadDataWithBaseURL("file:///android_asset/", html,
-            "text/html; charset=utf-8", "utf-8", null);
+                "text/html; charset=utf-8", "utf-8", null);
```

Comment:
This auto indentation is a configuration in IntelliJ. You can import the settings file: android-studio-settings.jar to solve it.